### DISCUSSION TITLE
Centrally handle the PointerEnteredEvent and PointerExitedEvent events?

### DISCUSSION THREAD

**Discussion by xiejiang2014** [AvaloniaUI/Avalonia#17845] [Q&A]
There are many child elements in a Canvas, and I need to centrally handle the PointerEnteredEvent and PointerExitedEvent events of these child elements.

However, I found that the PointerEnteredEvent and PointerExitedEvent events are registered as RoutingStrategies.Direct instead of RoutingStrategies.Bubble

class InputElement:
```
/// <summary>
/// Defines the <see cref="E:Avalonia.Input.InputElement.PointerEntered" /> event.
/// </summary>
public static readonly RoutedEvent<PointerEventArgs> PointerEnteredEvent = RoutedEvent.Register<InputElement, PointerEventArgs>("PointerEntered", RoutingStrategies.Direct);

/// <summary>
/// Defines the <see cref="E:Avalonia.Input.InputElement.PointerExited" /> event.
/// </summary>
public static readonly RoutedEvent<PointerEventArgs> PointerExitedEvent = RoutedEvent.Register<InputElement, PointerEventArgs>("PointerExited", RoutingStrategies.Direct);
```

How should I handle this situation? Thank you.

**Comment by stevemonaco** [comment]
You can do hit detection of children yourself on the `Canvas.PointerMoved` event and track the hovered control. This is sometimes the better option depending on what kind of app you're writing.

Otherwise, you probably need to write an attached behavior for `PointerEntered` + `PointerExited` and apply them to each child control. These will either: 1. publish a message that you can listen to or 2. crawl up the visual tree to find the appropriate control and call a method on it. You can apply these to every child via style selector.
